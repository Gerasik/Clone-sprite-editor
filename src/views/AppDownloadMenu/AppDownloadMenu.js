export default class AppDownloadMenu {
  constructor(className, parentBlock) {
    this.className = className;
    this.parentBlock = parentBlock;
  }

  render() {
    const downloadMenu = document.createElement('section');
    downloadMenu.classList.add(this.className);
    downloadMenu.innerHTML = `<span>Export:</span>
                              <button class="${this.className}__download-gif" title="download gif">to GIF</button>
                              <button class="${this.className}__download-jpeg" title="download jpeg">to JPEG</button>`;
    this.parentBlock.appendChild(downloadMenu);
    return downloadMenu;
  }
}
