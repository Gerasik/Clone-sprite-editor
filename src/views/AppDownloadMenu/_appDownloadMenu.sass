.download-menu
  padding: 20px 0 0
  font-size: 18px
  &__download-gif
    cursor: pointer
    &:hover
      background-color: #fff