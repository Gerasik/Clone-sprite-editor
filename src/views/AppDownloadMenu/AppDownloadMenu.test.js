import AppDownloadMenu from './AppDownloadMenu';

const content = {
  className: 'menu',
  parentBlock: document.body,
};

describe('new AppDownloadMenu', () => {
  const obj = new AppDownloadMenu('menu', document.body);
  it('Should be an instance of Object', () => {
    expect(obj).toBeInstanceOf(Object);
  });
  it('Should be contain right parameters', () => {
    expect(obj.className).toBe(content.className);
  });
});

describe('AppDownloadMenu.prototype.render', () => {
  it('Should be an instance of Function', () => {
    expect(AppDownloadMenu.prototype.render.bind(content)).toBeInstanceOf(Function);
  });

  it('Should be render slide', () => {
    expect(AppDownloadMenu.prototype.render.bind(content)).toMatchSnapshot();
  });
  const ans = AppDownloadMenu.prototype.render.bind(content)();
  it('Should be create 3 child', () => {
    expect(ans.children.length).toBe(3);
  });
});
