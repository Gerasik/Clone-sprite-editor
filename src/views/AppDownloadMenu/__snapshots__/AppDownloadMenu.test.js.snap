// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppDownloadMenu.prototype.render Should be render slide 1`] = `[Function]`;
